src/theory/strings/regexp_operation.cpp

```diff
@@ -791,8 +785,17 @@ void RegExpOpr::firstChars(Node r, std::set<unsigned> &pcset, SetNodes &pvset)
         firstChars(r[0], cset, vset);
         break;
       }
+      case kind::REGEXP_SIGMA:
       default: {
-        Assert(!isRegExpKind(k));
+        // should not call this function on regular expressions that aren't a
+        // standard regular expression kind.
+        Assert(k == REGEXP_SIGMA);
```

Comment:
Hmm, something is off here: we execute this code for both `case kind::REGEXP_SIGMA` and `default` but we assert here that it is only for `REGEXP_SIGMA`. It seems that we should just make this the `REGEXP_SIGMA` case and make the `default` case `Unreachable()`?

Reply:
That would change the behavior (in production).

The code here conservatively says any character can be the first character if we have sigma or an unknown operator. We additionally want to throw an assertion in debug, since we aren't expect to execute this for the latter.

Reply:
Hmm, I understand your reasoning but I am really not a huge fan of code that changes behavior in production because it makes it much more difficult to spot potential issues (i.e. everything may be fine for our users but actually, it just works by luck). If we keep this code as-is, we should at least comment on why this is safe to do for an unknown operator.

Reply:
The code is doing the right thing in production.  This is the common use case of a debug assertion (to proactively catch cases where we are executing something unexpected).

I can add a comment about why the behavior is correct for the default case.

Reply:
I've added a comment, let me know what you think.

Reply:
It is a common use case that I tend to disagree with :) However, the comment is good enough for me.

Reply:
I think I brought this up previously but I think that we should have a different macro for these cases, e.g. `Expect()` instead of `Assert()` to signal that we do the right thing even if the condition does not hold.